src/Network.js

```diff
@@ -480,18 +512,37 @@ class Network extends AsyncEventEmitter {
     try {
       const pdvs = pdu.getPdvs();
       pdvs.forEach((pdv) => {
-        if (!this.dimseBuffer) {
-          this.dimseBuffer = SmartBuffer.fromOptions({
-            encoding: 'ascii',
-          });
-        }
-        this.dimseBuffer.writeBuffer(pdv.getValue());
         const presentationContext = this.association.getPresentationContext(
           pdv.getPresentationContextId()
         );
+
+        if (!this.dimse) {
+          // Create stream to receive command
+          if (!this.dimseStream) {
+            this.dimseStream = MemoryStream.createWriteStream();
+            this.dimseStoreStream = undefined;
+          }
+        } else {
+          // Create stream to receive dataset
+          if (this.dimse.getCommandFieldType() === CommandFieldType.CStoreRequest) {
+            if (!this.dimseStoreStream) {
+              this.dimseStream = undefined;
+              this.dimseStoreStream = this.createStoreWritableStream(presentationContext);
+            }
+          } else {
+            if (!this.dimseStream) {
+              this.dimseStream = MemoryStream.createWriteStream();
+              this.dimseStoreStream = undefined;
+            }
+          }
+        }
+
+        const stream = this.dimseStream || this.dimseStoreStream;
+        stream.write(pdv.getValue());
```

Comment:
Here it should check the return value of write(). If it is not true, then we need to apply backpressure and until the drain event to resume. https://nodejs.org/docs/latest/api/stream.html#event-drain

Reply:
Nice catch! Thank you Kinson! The only thing that bothers me is how to wait in here for the `drain` event... any ideas?

Reply:
Any ideas on this @kinsonho?

Reply:
@PantelisGeorgiadis sorry for the late reply.

A pattern that I saw others doing is the following:

    if(!stream.write(pdv.getValue())) {
        await new Promise(resolve => stream.once('drain', resolve));
    }

Reply:
Thank you @kinsonho! The async/await pattern is not used in this library. Do you see any drawbacks in start using it (e.g. compatibility, etc.)?